File: sdk-java/dephealth-core/src/main/java/biz/kryukov/dev/dephealth/CheckResult.java
package biz.kryukov.dev.dephealth;

import java.util.Objects;

/**
 * Classification of a health check outcome: status category and detail.
 */
public record CheckResult(String category, String detail) {

    public CheckResult {
        Objects.requireNonNull(category, "category");
        Objects.requireNonNull(detail, "detail");
    }

    /** Successful check result. */
    public static final CheckResult OK = new CheckResult(StatusCategory.OK, StatusCategory.OK);
}

File: sdk-csharp/DepHealth.Core/CheckResult.cs
namespace DepHealth;

/// <summary>
/// Classification of a health check outcome: status category and detail.
/// </summary>
/// <param name="Category">The status category (e.g. "ok", "timeout", "error"). See <see cref="StatusCategory"/>.</param>
/// <param name="Detail">A more specific detail string describing the check outcome.</param>
public readonly record struct CheckResult(string Category, string Detail)
{
    /// <summary>Successful check result.</summary>
    public static readonly CheckResult Ok = new(StatusCategory.Ok, StatusCategory.Ok);
}
